test/generators/serializer_generator_test.rb

```diff
@@ -54,4 +53,21 @@ def test_with_no_attributes_does_not_add_extra_space
       end
     end
   end
+
+  private
+
+  def stub_safe_constantize(expected:)
+    String.class_eval do
+      alias_method :old, :safe_constantize
+    end
+    String.send(:define_method, :safe_constantize) do
+      Class if self == expected
+    end
+
+    yield
+  ensure
+    String.class_eval do
+      alias_method :safe_constantize, :old
+    end
+  end
```

Comment:
I don't think the `class << "ApplicationSerializer"` will work - gave it a shot and failed. Wouldn't that only patch the singleton of that particular *instance* of String?